hack/dockerfiles/lint.Dockerfile

```diff
@@ -3,7 +3,7 @@
 FROM golang:1.18-alpine
 ENV GOFLAGS="-buildvcs=false"
 RUN apk add --no-cache gcc musl-dev yamllint
-RUN wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.45.0
+RUN wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.47.3
```

Comment:
Had to upgrade this because with the new package upgrades, the 1.45 version was getting into a panic. See: https://github.com/moby/buildkit/runs/7755045750?check_suite_focus=true

Also several fixes in this newer version which also helped catch a call to a deprecated function in client/client.go